libs/utils/analysis/frequency_analysis.py

```diff
@@ -218,6 +218,107 @@ def plotClusterFrequencies(self, title='Clusters Frequencies'):
 
         return (avg_lfreq/1e3, avg_bfreq/1e3)
 
+    def plotCpusFrequencies(self, cpus=None):
+        """
+        Plot frequency trend for the specified CPUs (or all is not specified).
+        If sched_overutilized events ar available, the plots will also show the
+        intervals of time where the CPU was overutilized.
+
+        :param cpus: the list of CPUs to plot
+        :type cpus: int or list(int)
+        """
+        if not self._trace.hasEvents('cpu_frequency'):
+            self._log.warning('Events [cpu_frequency] not found, plot DISABLED!')
+            return
+        df = self._dfg_trace_event('cpu_frequency')
+
+        pd.options.mode.chained_assignment = None
+
+        freq = {}
+        for cpu_id in listify(cpus):
```

Comment:
If 'cpus' is unspecified, the list will only be [None]. plotCPUFrequencyResidency() already has something like that:

if cpus is None:
    # Generate plots only for available CPUs
    cpufreq_data = self._dfg_trace_event('cpu_frequency')
    _cpus = range(cpufreq_data.cpu.max()+1)
else:
    _cpus = listify(cpus)